Simplified/Reader2/Internal/ReaderModule.swift

```diff
@@ -14,120 +15,75 @@ import UIKit
 import R2Shared
 
 
-/// The ReaderModule handles the presentation of publications to be read by the
-/// user. It contains sub-modules implementing ReaderFormatModule to handle each
-/// format of publication (eg. CBZ, EPUB).
+/// The ReaderModule handles the presentation of publications to be read by the user.
+/// It contains sub-modules implementing ReaderFormatModule to handle each format of publication (eg. CBZ, EPUB).
 protocol ReaderModuleAPI {
-
-  var delegate: ReaderModuleDelegate? { get }
-
-
+  
+  var delegate: ModuleDelegate { get }
+  
   /// Presents the given publication to the user, inside the given navigation controller.
-  ///
   /// - Parameter publication: The R2 publication to display.
   /// - Parameter book: Our internal book model related to the `publication`.
-  /// - Parameter navigationController: The navigation stack the book will be
-  /// presented in.
+  /// - Parameter navigationController: The navigation stack the book will be presented in.
   /// - Parameter completion: Called once the publication is presented, or if an error occured.
-  func presentPublication(_ publication: Publication,
-                          book: NYPLBook,
-                          in navigationController: UINavigationController,
-                          completion: @escaping () -> Void)
-
-}
-
-protocol ReaderModuleDelegate: ModuleDelegate {
-
-  /// Called when the reader needs to load the R2 DRM object for the given publication.
-  func readerLoadDRM(for book: NYPLBook,
-                     completion: @escaping (CancellableResult<DRM?>) -> Void)
-
+  func presentPublication(publication: Publication, book: NYPLBook, in navigationController: UINavigationController, completion: @escaping () -> Void)
+  
 }
 
-
 final class ReaderModule: ReaderModuleAPI {
-
-  weak var delegate: ReaderModuleDelegate?
+  
+  var delegate: ModuleDelegate
```

Comment:
now though by removing `weak` I think we have a retain cycle, since NYPLR2Owner "is" the ModuleDelegate, but it also owns the ReaderModule instance. So we need to break the cycle in some way.
```suggestion
  var delegate: ModuleDelegate
```
```suggestion
  var delegate: ModuleDelegate
```
```suggestion
  var delegate: ModuleDelegate
```
```suggestion
  var delegate: ModuleDelegate
```
```suggestion
  var delegate: ModuleDelegate
```
```suggestion
  var delegate: ModuleDelegate
```
```suggestion
  var delegate: ModuleDelegate
```
```suggestion
  var delegate: ModuleDelegate
```

Reply:
hmm you're right... I'll keep it optional then. added logging in case it is nil